contrib/pkg/awsprivatelink/enable.go

```diff
@@ -0,0 +1,224 @@
+package awsprivatelink
+
+import (
+	"context"
+	"fmt"
+	"os/user"
+	"path/filepath"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+
+	hivev1 "github.com/openshift/hive/apis/hive/v1"
+	hiveutils "github.com/openshift/hive/contrib/pkg/utils"
+	awsutils "github.com/openshift/hive/contrib/pkg/utils/aws"
+	hiveclient "github.com/openshift/hive/pkg/client/clientset/versioned"
+
+	openshiftcorev1 "github.com/openshift/client-go/config/clientset/versioned/typed/config/v1"
+	"github.com/openshift/hive/pkg/constants"
+
+	"github.com/google/uuid"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+type enableOptions struct {
+	homeDir string
+	// AWS region of the Hive cluster
+	region        string
+	infraId       string
+	dnsRecordType string
+
+	kubernetesClient      *kubernetes.Clientset
+	hiveClient            *hiveclient.Clientset
+	openshiftConfigClient *openshiftcorev1.ConfigV1Client
+
+	// AWS EC2 client in Hive cluster's region
+	EC2Client *ec2.EC2
+}
+
+func NewEnableAWSPrivateLinkCommand() *cobra.Command {
+	opt := &enableOptions{}
+
+	cmd := &cobra.Command{
+		Use:   "enable",
+		Short: "Enable AWS PrivateLink",
+		Long: `Enable AWS PrivateLink:
+1) Extract AWS Hub account credentials from the environment where this command is called
+2) Create a Secret with the above credential. The Secret is referred to by HiveConfig.spec.awsPrivateLink.credentialsSecretRef
```

Comment:
This is slightly confusing: I can't tell by reading it whether the command will add the credentialsSecretRef or if that's my responsibility. Reading the code I see it's the former, which is cool. Suggest splitting the second sentence into its own bullet and rewording slightly, e.g.
```
3) Add a reference to the Secret in HiveConfig.spec.awsPrivateLink.credentialsSecretRef.
```

Reply:
Fixed.